maintenance/health/servicehealthcheck/healthchecker.go

```diff
@@ -68,13 +66,55 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Checks one health check
+func check(hc HealthCheck, name string) (error, bool) {
+	// If it was not possible to initialise this health check, then show the initialisation error
+	if val, isIn := initErrors.Load(name); isIn {
+		err := val.(error)
+		return err, false
+	}
+	// this is the actual health check
+	healthy, err := hc.HealthCheck()
+	return err, healthy
+}
+
+// Does all HealthChecks and lists the result in the response body
+func checkAll(w http.ResponseWriter) {
```

Comment:
You have a function `check()` and `checkAll()`, which look very similar, but ultimately behave totally different. What bothers me the most is that the one is writing the response directly and the other does not. And I think @Ferlonas' idea to generalise checking one/many was a good idea. And it should make it easy to separate checking and response generation. Like `ServeHTTP()` does all the response generation and it looks somewhat like this:
```go
checks := allChecks
if name != "all" {
	checks = map[string]HealthCheck{name: allChecks[name]} // but obviously check that allChecks[name] is defined
}
result := check(checks) // func check(map[string]HealthCheck) map[string]error
if name != "all" {
	// write result[name]
} else {
	for name, hc := range result {...}
}
```